In [1]:
import pandas as pd

df=pd.read_csv("../data/application_train.csv")
print(df.head())
print(" ")
print(df.tail())

   SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0      100002       1         Cash loans           M            N   
1      100003       0         Cash loans           F            N   
2      100004       0    Revolving loans           M            Y   
3      100006       0         Cash loans           F            N   
4      100007       0         Cash loans           M            N   

  FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
0               Y             0          202500.0    406597.5      24700.5   
1               N             0          270000.0   1293502.5      35698.5   
2               Y             0           67500.0    135000.0       6750.0   
3               Y             0          135000.0    312682.5      29686.5   
4               Y             0          121500.0    513000.0      21865.5   

   ...  FLAG_DOCUMENT_18 FLAG_DOCUMENT_19 FLAG_DOCUMENT_20 FLAG_DOCUMENT_21  \
0  ...                 0             

In [2]:
print(df.shape)

(307511, 122)


### Observation: Dataset Shape

- The dataset contains **307,511 loan applications**.
- It consists of **122 attributes**, including demographic, financial, credit history, and behavioral information.
- The target variable is `TARGET`, which indicates whether the applicant experienced payment difficulties (1) or repaid the loan successfully (0).

In [3]:
print(df.columns)

Index(['SK_ID_CURR', 'TARGET', 'NAME_CONTRACT_TYPE', 'CODE_GENDER',
       'FLAG_OWN_CAR', 'FLAG_OWN_REALTY', 'CNT_CHILDREN', 'AMT_INCOME_TOTAL',
       'AMT_CREDIT', 'AMT_ANNUITY',
       ...
       'FLAG_DOCUMENT_18', 'FLAG_DOCUMENT_19', 'FLAG_DOCUMENT_20',
       'FLAG_DOCUMENT_21', 'AMT_REQ_CREDIT_BUREAU_HOUR',
       'AMT_REQ_CREDIT_BUREAU_DAY', 'AMT_REQ_CREDIT_BUREAU_WEEK',
       'AMT_REQ_CREDIT_BUREAU_MON', 'AMT_REQ_CREDIT_BUREAU_QRT',
       'AMT_REQ_CREDIT_BUREAU_YEAR'],
      dtype='str', length=122)


### Observation: Dataset Features

- The dataset contains **122 attributes** describing loan applicants.
- Features include demographic information, income, employment, education, housing, family details, credit amount, annuity, external credit scores, and bureau inquiry history.
- The dataset also contains a binary target variable (`TARGET`) indicating whether the applicant experienced payment difficulties.
- `SK_ID_CURR` serves as a unique identifier for each loan application and will not be used as a predictive feature.

In [4]:
print(df.info())

<class 'pandas.DataFrame'>
RangeIndex: 307511 entries, 0 to 307510
Columns: 122 entries, SK_ID_CURR to AMT_REQ_CREDIT_BUREAU_YEAR
dtypes: float64(65), int64(41), str(16)
memory usage: 286.2 MB
None


### Dataset Information

- The dataset contains **307,511 loan applications** with **122 features**.
- The dataset occupies approximately **286.2 MB** of memory.
- There are **65 floating-point**, **41 integer**, and **16 categorical (string)** features.
- The presence of multiple data types indicates that preprocessing techniques such as missing value handling, categorical encoding, and feature scaling will be required before model training.